third_party/nvfuser/test/test_gpu2.cpp

```diff
@@ -9028,27 +9028,27 @@ TEST_F(NVFuserTest, FusionChannelsLastParser_CUDA) {
   // 2. use a fuzzy compare (ignore non-significant whitespaces for example)
   const std::string expected_kernel = R"(
 __global__ void CUDAGeneratedKernel(Tensor<__half, 4> T0, Tensor<__half, 4> T2, Tensor<__half, 4> T7) {
-  int64_t i295;
-  i295 = T0.size[2] * T0.size[1];
-  int64_t i298;
-  i298 = ((nvfuser_index_t)threadIdx.x) + (128 * ((nvfuser_index_t)blockIdx.x));
-  int64_t i300;
-  i300 = (T0.size[1] * T0.size[2]) * T0.size[3];
-  int64_t i332;
-  i332 = i298 % i300;
-  int64_t i309;
-  i309 = T0.size[2] * T0.size[3];
-  int64_t i333;
-  i333 = i332 % i309;
-  if ((i298 < (((T0.size[0] * T0.size[1]) * T0.size[2]) * T0.size[3]))) {
+  int64_t i558;
```

Comment:
It is annoying to update this string comparison everytime when there is an expr simplifier change. I will write a diff algorithm and embed it into `assertCUDAKernel`, and allow passing when there is only trivial var name change.

Reply:
I don't think we should just always ignore string changes because sometimes there could be something that would need a real attention. I don't know if we could automatically say if a string change would be safe to ignore. At least, it would be great if we could expand the string comparison with a semantic comparison that would kick in when the string comparison failed. We should still throw an error when a string mismatch is detected, but it would be greatly helpful if the error message could indicate if the error is just superficial or not.